src/main/scala/mesosphere/mesos/Availability.scala

```diff
@@ -0,0 +1,23 @@
+package mesosphere.mesos
+
+import org.apache.mesos.Protos.Offer
+
+object Availability {
+
+  def offerAvailable(offer: Offer): Boolean = {
+    var unavailability = offer.hasUnavailability
+    var now = System.nanoTime()
+    if (offer.hasUnavailability && offer.getUnavailability.hasStart) {
+      val start = offer.getUnavailability.getStart.getNanoseconds
+      if (now.>=(start)) {
+        if (offer.getUnavailability.hasDuration) {
+          return start.+(offer.getUnavailability.getDuration.getNanoseconds).<(now)
```

Comment:
It will be cool if you allow configuring a period in which offer will be declined. For example if agent will be unavailable in next 15 minutes then there is no point in starting task there.